### DISCUSSION TITLE
Restore database from backup in swarm

### DISCUSSION THREAD

**Discussion by abanoubadel** [BretFisher/ama#184] [Q&A]
Hi Bret,

I'm having challenge of how to restore database (mysql) from backup in a production docker swarm. I'm triggering/creating the restore process from Jenkins in docker service called 'restore_service'. The challenge is how to know that the restore_service is finished, so that I can remove it.

After the restore script is finished the container exits, however docker swarm will launch it again. and the restore script will start again. I wish if there was the `--rm` option to docker service. 

I'm thinking of sharing the docker manager socket inside 'restore_service' and perform docker service rm from inside. Still don't know how to do that yet.

Can't wait to your answer of how to solve this challenge. and if there is any other solution.
Thanks for your effort.

**Comment by BretFisher** [comment]
Try setting the restore_service `restart_policy:` to `none`, then manually `service rm` it later. Does that work?

**Comment by abanoubadel** [comment]
@BretFisher  Yesssssss it works, Thanks a lot.
